students/ghassan/lesson06/test_new_mailroom.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+import unittest
+from new_mailroom import main_menu, send_thankyou, donors
+
+
+class TestMailroom(unittest.TestCase):
+    def test_main_menu(self):
+        ''' this tests if the function is actually
+        returning the user's input '''
+        users_choice = main_menu()
+        self.assertEqual(users_choice, users_choice)
+
+    def test_send_thankyou_new_donor(self):
+        ''' tests whether a new donor was actually added
+        to the donors list'''
+        # I did a list comprehention here to get
+        # a fresh copy of the dictionary keys as of
+        # a reference or a pointer to the same dictionary
+        old_donors = [donor for donor in donors.keys()]
+        send_thankyou()
```

Comment:
Testing in this way would require you to enter user input via the command line during your tests which defeats the purpose of automated testing. Can you think of a way to refactor your `send_thankyou` function to make it more testable? I would suggest isolating letter generation into a separate function that returns rather than prints the letter so that it can be tested. I would suggest having your `create_report` function return a value that can be printed, making it more testable as well.